pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -616,6 +621,17 @@ public void run() {
     }
   }
 
+  // Upload the tar file of the committed segment to the configure segment store.
+  private boolean uploadSegmentToSegmentStore(File segmentTarFile) {
+    if (segmentTarFile == null) {
+      return false;
+    }
+    if (!segmentTarFile.exists()) {
+      segmentLogger.warn("File does not exist: {}", segmentTarFile.getAbsolutePath());
```

Comment:
Shouldn't this throw an exception instead of just warn ? Better yet, should we use a precondition here ?